pages/wiki.js

```diff
@@ -0,0 +1,144 @@
+import Head from 'next/head'
+import Navbar from '../components/Navbar';
+import Fuse from 'fuse.js';
+import links from '../links.json';
+import React, { useState } from 'react';
+
+function wiki() {
+
+  const [query, setQuery] = useState('');
+
+  const fuse = new Fuse(links, {
+    keys: [
+      'title',
+      'description',
+      'type'
+    ]
+  })
```

Comment:
Hmm.. I'm not entirely sure how often this will get created (it may or may not be a lot, I need to brush up on my React hook knowledge), but I'd rather do "state management" outside of this file. Let's pluck this out into its own thing:

so, create a new file, something like `store/fuse.js`, and move it there. In that file, we should be exporting this `fuse` object. And in this file, we can import it. This way, we'll be sure only to have one instance of this index throughout our application.

Let me know if this doesn't make any sense

Reply:
I'm not exactly the most solid on react hooks either. The React docs state that React hooks can be used in function( ) based components. This operation relies on internal variables already, it might be possible to store those variables and operations in a separate function( ) based component.

Reply:
I'm not sure if I'm following what you're thinking

Reply:
Put together a little Loom to show you what I'm talking about: https://www.loom.com/share/8b0c2dca8f704045862538855a316be5

Reply:
got it, changes are on the way